### PR TITLE
Fix check links and make it simpler

### PR DESCRIPTION
**Pull request by pipo02mix**
### What this PR does / why we need it

It split the check link validation for doc section making it easier to track errors. Also it reduces the checks in `changes/` section since we don't want to fix external URLs for old changelog entries

### Things to check/remember before submitting

- If you made content changes

    - Run `make dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

File: .github/workflows/check-links-in-prod.yaml [hunk 1]
```diff
@@ -12,15 +12,54 @@ jobs:
...
+        run: |
+          docker run --rm --name linkchecker \
+            ghcr.io/linkchecker/linkchecker:latest \
+            https://docs.giantswarm.io/changes \
+            -t 1 -r 2 \
+            --ignore-url="^https://github.com/giantswarm/docs/.*" \
+            --ignore-url="^https://.*example\.com/.*" \
+            --ignore-url="^https://my-org\.github\.com/.*" \
+            --ignore-url="^https://github\.com/giantswarm/giantswarm/.*" \
+            --ignore-url=".*gigantic\.io.*"
```
Comment (pipo02mix) [inline .github/workflows/check-links-in-prod.yaml:65]: This makes it easier to consume the output and does not fall into the whole of deep validation (we do have not control over when the old changelog points to a non-existing URL)

**Comment by marians** [comment]
It seems like this is not working as intended. I ran it in https://github.com/giantswarm/docs/actions/runs/8539886897/job/23395648737 and the first check starts on the /overview page, but recurses into all other pages. So we end up checking the entire site multiple times.

File: .github/workflows/check-links-in-prod.yaml [hunk 2]
```diff
@@ -12,15 +12,54 @@ jobs:
...
+            -t 1
+
+      - name: Check links in reference pages
+        run: |
+          docker run --rm --name linkchecker \
+            ghcr.io/linkchecker/linkchecker:latest \
+            https://docs.giantswarm.io/reference \
+            --check-extern \
+            -t 1
+
```
Comment (pipo02mix) [inline .github/workflows/check-links-in-prod.yaml:46]:
```suggestion
            https://docs.giantswarm.io/overview/ \
            --check-extern \
            -t 1 -r 2

      - name: Check links in getting-started pages
        run: |
          docker run --rm --name linkchecker \
            ghcr.io/linkchecker/linkchecker:latest \
            https://docs.giantswarm.io/getting-started/ \
            --check-extern \
            -t 1-r 2

      - name: Check links in tutorials pages
        run: |
          docker run --rm --name linkchecker \
            ghcr.io/linkchecker/linkchecker:latest \
            https://docs.giantswarm.io/tutorials/ \
            --check-extern \
            -t 1 -r 2

      - name: Check links in reference pages
        run: |
          docker run --rm --name linkchecker \
            ghcr.io/linkchecker/linkchecker:latest \
            https://docs.giantswarm.io/reference/ \
            --check-extern \
            -t 1 -r 2

```

File: .github/workflows/check-links-in-prod.yaml [hunk 3]
```diff
@@ -12,15 +12,54 @@ jobs:
...
+            https://docs.giantswarm.io/support \
+            --check-extern \
+            -t 1
+
+      - name: Check links in changelogs
+        run: |
+          docker run --rm --name linkchecker \
+            ghcr.io/linkchecker/linkchecker:latest \
+            https://docs.giantswarm.io/changes \
+            -t 1 -r 2 \
```
Comment (pipo02mix) [inline .github/workflows/check-links-in-prod.yaml:60]:
```suggestion
      - name: Check links in support pages
        run: |
          docker run --rm --name linkchecker \
            ghcr.io/linkchecker/linkchecker:latest \
            https://docs.giantswarm.io/support/ \
            --check-extern \
            -t 1 -r 2

      - name: Check links in changelogs
        run: |
          docker run --rm --name linkchecker \
            ghcr.io/linkchecker/linkchecker:latest \
            https://docs.giantswarm.io/changes/ \
            -t 1 -r 2 \
```

**Comment by pipo02mix** [comment]
[Fixed](https://github.com/giantswarm/docs/actions/runs/8554223183/job/23439085880). I will try to pass the tests too

File: .github/workflows/check-links-in-prod.yaml [hunk 4]
```diff
@@ -12,15 +12,54 @@ jobs:
...
+            --check-extern \
+            -t 1 -r 2
+
+      - name: Check links in getting-started pages
+        run: |
+          docker run --rm --name linkchecker \
+            ghcr.io/linkchecker/linkchecker:latest \
+            https://docs.giantswarm.io/getting-started/ \
+            --check-extern \
+            -t 1-r 2
```
Comment (marians) [inline .github/workflows/check-links-in-prod.yaml:29]:
```suggestion
            -t 1 -r 2
```

**Comment by pipo02mix** [comment]
sadly it runs against the current online docs.giantswarm.io page so I cannot demonstrate the changes fix the linter :)

**Comment by pipo02mix** [comment]
@marians can you take a look?

**Comment by marians** [comment]
I just ran the workflow here: https://github.com/giantswarm/docs/actions/runs/8615042101/job/23609717739

The first step finds broken links and exits with non-zero exit code. As a result, the workflow does not continue after the step, so other links are not checked.

**Comment by pipo02mix** [comment]
yeah that is a problem, maybe I can cache all errors and only stop if the array is not empty, let me try

**Comment by pipo02mix** [comment]
now should be okish @marians PTAL

**Comment by marians** [comment]
Looks like we are getting there!

`::set-output` is deprecated and will be removed. See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ for info on what to use instead.

**Comment by marians** [comment]
Not sure where I saw `::set-output`. Probably looked at an outdated diff.

**Review by marians — APPROVED** [review]
LGTM so far. Ideally let's switch to something else than `::set-output

**Pull request merged**